### PR TITLE
feat(autoplay-gallery): add autoplay to gallery in showcase mode

### LINKED ISSUE #638: Presenter mode: gallery
- [ ] play through gallery in show case mode
- [ ] in presenter mode clicks should progress through gallery before moving on to next slide.

### PR DESCRIPTION
**Pull request by KatvonRivia**
closes #638

### CODE REVIEW

File: src/scripts/hooks/use-story-navigation.ts [hunk 1]
```diff
@@ -35,7 +38,12 @@ export const useStoryNavigation = () => {
   if (storyIds && typeof storyIndex === 'number') {
     const showcaseStoryIds = storyIds.join('&');
     const nextStoryIndex = storyIndex + 1;
+    const currentSlide = selectedStory?.slides[slideIndex];
     // go through all slides of one story
+
+    if (currentSlide?.images && currentSlide.type === SlideType.Image) {
+      delay = delay * currentSlide.images.length;
+    }
```
Comment (pwambach) [inline src/scripts/hooks/use-story-navigation.ts:46]: add empty line

File: src/scripts/hooks/use-story-navigation.ts [hunk 2]
```diff
@@ -15,6 +17,7 @@ export const useStoryNavigation = () => {
   let autoPlayLink = null;
   let nextSlideLink = null;
   let previousSlideLink = null;
+  let delay = 3000;
```
Comment (pwambach) [inline src/scripts/hooks/use-story-navigation.ts:20]: Would be cool if this value is defined in the config file

File: src/scripts/components/stories/story-media/story-media.tsx [hunk 3]
```diff
@@ -27,6 +32,15 @@ const StoryMedia: FunctionComponent<Props> = ({
   const showPrevButton = currentIndex > 0;
   const showNextButton = currentIndex < images.length - 1;
 
+  useInterval(() => {
```
Comment (pwambach) [inline src/scripts/components/stories/story-media/story-media.tsx:35]: This way the interval function is called every 3 seconds also in non showcase stories. Better put the `mode === StoryMode.Showcase` check into the delay. If the delay is `null` the function will not be triggered.

Like:
```
delay = mode === StoryMode.Showcase ? 3000 : null
```

File: src/scripts/components/stories/story-media/story-media.tsx [hunk 4]
```diff
@@ -6,19 +6,24 @@ import {NextIcon} from '../../main/icons/next-icon';
...
 import styles from './story-media.styl';
 
 interface Props {
   images: string[];
   imageCaptions?: string[];
   storyId: string;
+  mode: StoryMode | null;
 }
 
 const StoryMedia: FunctionComponent<Props> = ({
```
Comment (pwambach) [inline src/scripts/components/stories/story-media/story-media.tsx:22]: Another thing but not related to this PR. It's a bit confusing that this component is called StoryMedia and not sth like StoryGallery or StoryImages. At first I thought StoryMedia is a wrapper component for images and videos etc.

**Review by pwambach — CHANGES_REQUESTED** [review]

File: src/scripts/components/stories/story-media/story-media.tsx (hunk 4 above)
Comment (KatvonRivia) [reply src/scripts/components/stories/story-media/story-media.tsx:23]: Good point, will rename it in a different PR. 👍

**Review by pwambach — APPROVED** [review]

**Pull request merged**